clients/intellij/src/main/kotlin/com/tabbyml/intellijtabby/inlineChat/InlineChatIntentionAction.kt

```diff
@@ -0,0 +1,409 @@
+        layout = BorderLayout()
+        add(historyButton, BorderLayout.WEST)
+        add(textArea, BorderLayout.CENTER)
+        add(submitButton, BorderLayout.EAST)
+        border = BorderFactory.createLineBorder(UIUtil.getHeaderInactiveColor(), 2, true)
+
+        addKeyListener(object : KeyListener {
+            override fun keyPressed(e: KeyEvent) {
+                e.consume()
+            }
+
+            override fun keyReleased(e: KeyEvent) {
+                if (e.keyCode == KeyEvent.VK_BACK_SPACE || e.keyCode == KeyEvent.VK_DELETE) {
+                    e.consume()
+                }
+            }
+
+            override fun keyTyped(e: KeyEvent) {
+                e.consume()
+            }
+        })
+    }
+
+    private fun createTextArea(): JBTextArea {
+        val textArea = JBTextArea().apply {
+            font = Font(font.family, font.style, 14)
+        }
+        textArea.lineWrap = false
+        textArea.rows = 1
+        textArea.columns = 30
+        textArea.emptyText.text = "Enter the command to editing"
+        textArea.border = BorderFactory.createEmptyBorder(6, 4, 4, 4)
+        // To prevent keystrokes(backspace, delete) being handled by the host editor, https://intellij-support.jetbrains.com/hc/en-us/community/posts/360010505760-Issues-embedding-editor-in-block-inlay
+        textArea.putClientProperty(UIUtil.HIDE_EDITOR_FROM_DATA_CONTEXT_PROPERTY, true)
+        textArea.addKeyListener(object : KeyListener {
+            override fun keyPressed(e: KeyEvent) {
+                //
+            }
+
+            override fun keyReleased(e: KeyEvent) {
+                if (e.keyCode == KeyEvent.VK_BACK_SPACE || e.keyCode == KeyEvent.VK_DELETE) {
+                    e.consume()
+                }
+
+                if (e.keyCode == KeyEvent.VK_ENTER) {
+                    handleConfirm()
+                }
+
+                if (e.keyCode == KeyEvent.VK_ESCAPE) {
+                    // Handle escape
+                    textArea.text = ""
+                    onCancel()
+                }
+            }
+
+            override fun keyTyped(e: KeyEvent) {
+                //
+            }
+        })
+        textArea.addFocusListener(object : FocusAdapter() {
+            override fun focusGained(e: FocusEvent) {
+                border = BorderFactory.createLineBorder(UIUtil.getFocusedBorderColor(), 2, true)
+            }
+
+            override fun focusLost(e: FocusEvent) {
+                border = BorderFactory.createLineBorder(UIUtil.getHeaderInactiveColor(), 2, true)
+            }
+        })
+        return textArea
+    }
+
+    override fun requestFocus() {
+        textArea.requestFocus()
+    }
+
+    private fun handleConfirm() {
+        onSubmit(textArea.text.trim())
+        textArea.text = ""
+    }
+
+    private fun createSubmitButton(): JLabel {
+        val submitButton = IconLabelButton(AllIcons.Actions.Commit) { handleConfirm() }
```

Comment:
```suggestion
        val submitButton = IconLabelButton(AllIcons.Chooser.Right) { handleConfirm() }
```
It would be good to use a right arrow here to indicate sending the command.